python/ray/data/read_api.py

```diff
@@ -2282,21 +2283,24 @@ def from_tf(
 @PublicAPI
 def from_torch(
     dataset: "torch.utils.data.Dataset",
+    parallelism: int = -1,
+    shuffle: bool = False,
 ) -> MaterializedDataset:
     """Create a :class:`~ray.data.Dataset` from a
     `Torch Dataset <https://pytorch.org/docs/stable/data.html#torch.utils.data.Dataset/>`_.
 
-    This function is inefficient. Use it to read small datasets or prototype.
-
     .. warning::
         If your dataset is large, this function may execute slowly or raise an
         out-of-memory error. To avoid issues, read the underyling data with a function
         like :meth:`~ray.data.read_images`.
 
     .. note::
-        This function isn't parallelized. It loads the entire dataset into the head
+        For iterable-style datasets, This function is not parallelized. It loads the entire dataset into the head
```

Comment:
let's also clarify that this implementation is now a streamed implementation, so can read larger than memory datasets